### PR TITLE
Purge Package Cloud snapshot RPMS

### PR DESCRIPTION
**Pull request by indigo423**
RPMS snapshots are released for every build on master and piles up. To have only the last working snapshot deployed the package cloud repository needs to be cleaned up. The following changes where made to get delete all SNAPSHOT packages:

* Centralize configuration for package cloud repository environment in .circleci/env/packagecloud
* Created a script which uses the ReST API to fetch all SNAPSHOT release packages and delete them via REST
* Add a yank-snapshot-packages build step into the build pipeline

### CODE REVIEW

**Comment by indigo423** [comment]
@j-white Can I add this branch to filter so we can see if this this works?

File: .circleci/scripts/package-cloud-yank-snapshots.sh [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  echo "The environment variable PACKAGECLOUD_REPO is not set and is required."
+  echo "Set PACKAGECLOUD_REPO environment variable."
+  exit 1
+fi
+
+BASE_URL="https://${PACKAGECLOUD_TOKEN}@packagecloud.io"
+
+SNAPSHOT_PACKAGE_URLS=$(${CURL_CMD} -s "${BASE_URL}/api/v1/repos/${PACKAGECLOUD_USER}/${PACKAGECLOUD_REPO}/packages.json" | ${JQ_CMD} '.[].destroy_url')
+
+for PKG_URL in ${SNAPSHOT_PACKAGE_URLS}; do
```
Comment (j-white) [inline .circleci/scripts/package-cloud-yank-snapshots.sh]: I don't think we can simply delete all of the packages - we may want to share this repository with other projects.

Also if we delete *all* of the snapshots *before* we push out the next out, then there will be some (small) period of time where there are no packages at all.

**Comment by j-white** [comment]
Good find for `packages.json`.

**Comment by j-white** [comment]
> @j-white Can I add this branch to filter so we can see if this this works?

Sure

**Pull request merged**